addons/sale/models/account_move_line.py

```diff
@@ -112,6 +116,9 @@ def _sale_create_reinvoice_sale_line(self):
                         # store it in the map_move_sale_line map
                         map_move_sale_line[move_line.id] = len(sale_line_values_to_create) - 1  # save the index of the value to create sale line
 
+            elif move_line.move_id.move_type in ('out_refund', 'in_refund'):
+                map_move_sale_line[move_line.id] = sale_order.order_line.filtered(lambda l: l.qty_delivered_method == 'analytic' and l.product_id == move_line.product_id)
```

Comment:
You need to take only the first one to avoid tracebacks (corner case, but managed with a `limit=1` above too...)